internal/events/tailer.go

```
@@ -0,0 +1,91 @@
+// sent to events; lines that fail ParseLine are sent to invalid (non-blocking
+// drop if the channel is full). Exits when ctx is cancelled. Designed to be
+// run in a goroutine.
+//
+// Partial lines (no trailing newline) are buffered until the newline arrives,
+// both during the live run and across reads. Emit only on newline.
+func Tail(ctx context.Context, path string, events chan<- Event, invalid chan<- string) {
+	f, err := os.Open(path)
+	if err != nil {
+		return
+	}
+	defer f.Close()
```

golangci-lint (2.11.4)

[error] Error return value of `f.Close` is not checked

(errcheck)